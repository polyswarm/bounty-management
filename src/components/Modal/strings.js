const strings = {
  ok: 'OK',
};

export default strings;
