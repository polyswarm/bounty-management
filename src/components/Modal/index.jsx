// Vendor imports
import React, { Component } from 'react';
import PropTypes from 'prop-types';
// Bounty imports
import Button from '../Button';
// Component imports
import strings from './strings';

class Modal extends Component {
  constructor(props) {
    super(props);
    this.state = {
      open: false,
    };

    this.onCloseClick = this.onCloseClick.bind(this);
    this.open = this.open.bind(this);
    this.close = this.close.bind(this);
  }

  render() {
    const { props: { message, title } } = this;
    const { state: { open } } = this;
    return (
      <div className='Modal'>
        {open && (
          <React.Fragment>
            <div className='ModalBackground'>
            </div>
            <div className='ModalContent'>
              <header className='ModalContentHeader'>
                {title}
              </header>
              <p>{message}</p>
              <span className='Modal-Button-Bar'>
                <Button
                  flat
                  onClick={this.onCloseClick}>
                  {strings.ok}
                </Button>
              </span>
            </div>
          </React.Fragment>
        )}
      </div>
    );
  }

  onCloseClick() {
    this.close();
  }

  open() {
    this.setState({open: true});
  }

  close() {
    this.setState({open: false});
  }
}

Modal.proptypes = {
  title: PropTypes.string,
};
export default Modal;
