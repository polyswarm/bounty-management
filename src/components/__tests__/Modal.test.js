import React from 'react';
import {render, mount} from 'enzyme';
import {renderToJson, mountToJson} from 'enzyme-to-json';
import Modal from '../Modal';

it('renders without crashing', () => {
  const wrapper = render(<Modal/>);

  expect(renderToJson(wrapper)).toMatchSnapshot();
});

it('opens without crashing', () => {
  const wrapper = mount(<Modal />);
  wrapper.setState({open: true});
  expect(mountToJson(wrapper)).toMatchSnapshot();
});

it('sets state open to true when open is called', () => {
  const wrapper = mount(<Modal/>);
  const instance = wrapper.instance();
  const setState = jest.spyOn(Modal.prototype, 'setState');

  instance.open();

  expect(setState).toHaveBeenCalledTimes(1);
  expect(setState).toHaveBeenCalledWith({open: true});
});

it('sets state open to false when close is called', () => {
  const setState = jest.spyOn(Modal.prototype, 'setState');
  const wrapper = mount(<Modal />);
  const instance = wrapper.instance();
  setState.mockClear();

  instance.close();

  expect(setState).toHaveBeenCalledTimes(1);
  expect(setState).toHaveBeenCalledWith({open: false});
});

it('hides the Modal when closed', () => {
  const wrapper = mount(<Modal />);
  wrapper.setState({open: false});

  expect(wrapper.find('.ModalBackground')).toHaveLength(0);
});

it('shows the Modal when open', () => {
  const wrapper = mount(<Modal />);
  wrapper.setState({open: true});

  expect(wrapper.find('.ModalBackground')).toHaveLength(1);
});

it('does not close the modal on click outside the main content', () => {
  const wrapper = mount(<Modal/>);
  wrapper.setState({open: true});

  wrapper.find('.ModalBackground').simulate('click');

  expect(wrapper.find('.ModalBackground')).toHaveLength(1);
});

it('shows the property title', () => {
  const title = 'Title';
  const wrapper = mount(<Modal title={title}/>);
  wrapper.setState({open: true});

  expect(wrapper.find('.ModalContentHeader')).toHaveLength(1);
  expect(wrapper.find('.ModalContentHeader').text()).toEqual('Title');
});

it('shows the property message', () => {
  const title = 'Title';
  const message = 'message';
  const wrapper = mount(<Modal
    title={title}
    message={message}/>
  );
  wrapper.setState({open: true});

  expect(wrapper.find('.ModalContent')).toHaveLength(1);
  expect(wrapper.find('.ModalContent').find('p').text()).toEqual('message');
});
