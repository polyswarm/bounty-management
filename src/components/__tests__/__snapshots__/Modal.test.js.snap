// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`opens without crashing 1`] = `
<Modal>
  <div
    className="Modal"
  >
    <div
      className="ModalBackground"
    />
    <div
      className="ModalContent"
    >
      <header
        className="ModalContentHeader"
      />
      <p />
      <span
        className="Modal-Button-Bar"
      >
        <Button
          flat={true}
          onClick={[Function]}
        >
          <button
            className="Button flat"
            onClick={[Function]}
          >
            OK
          </button>
        </Button>
      </span>
    </div>
  </div>
</Modal>
`;

exports[`renders without crashing 1`] = `
<div
  class="Modal"
>
  
</div>
`;
